src/renderer/coremods/settings/pages/General.tsx

```diff
@@ -110,56 +124,31 @@ export const General = (): React.ReactElement => {
 
         <SwitchItem
           value={expValue}
-          onChange={(value) => {
-            void modal
-              .confirm({
-                title: Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS,
-                body: Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS_DESC.format(),
-                confirmText: Messages.REPLUGGED_RELOAD,
-                confirmColor: Button.Colors.RED,
-              })
-              .then((answer) => {
-                if (answer) {
-                  expOnChange(value);
-                  reload();
-                }
-              });
-          }}
+          onChange={(value) => restartModal(false, () => expOnChange(value))}
           note={Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS_DESC.format()}>
           {Messages.REPLUGGED_SETTINGS_DISCORD_EXPERIMENTS}
         </SwitchItem>
 
         <SwitchItem
           value={rdtValue}
           onChange={(value) => {
-            void modal
-              .confirm({
-                title: Messages.REPLUGGED_SETTINGS_REACT_DEVTOOLS,
-                body: Messages.REPLUGGED_SETTINGS_REACT_DEVTOOLS_DESC.format(),
-                confirmText: Messages.REPLUGGED_RELOAD,
-                confirmColor: Button.Colors.RED,
-              })
-              .then((answer) => {
-                if (answer) {
-                  if (value) {
-                    void RepluggedNative.reactDevTools
-                      .downloadExtension()
-                      .then(() => {
-                        rdtOnChange(value);
-                        relaunch();
-                      })
-                      .catch(() => {
-                        void toast.toast(
-                          Messages.REPLUGGED_SETTINGS_REACT_DEVTOOLS_FAILED,
-                          toast.Kind.FAILURE,
-                        );
-                      });
-                  } else {
-                    rdtOnChange(value);
-                    relaunch();
-                  }
-                }
-              });
+            if (value) {
+              void RepluggedNative.reactDevTools
+                .downloadExtension()
+                .then(() => {
+                  rdtOnChange(value);
```

Comment:
Do the onChange before downloading so the switch doesn't flicker, but revert the change in the catch block.